src/live_restore/live_restore_server.c

```diff
@@ -22,6 +22,55 @@ __live_restore_worker_check(WT_SESSION_IMPL *session)
     return (true);
 }
 
+/*
+ * __live_restore_clean_up --
+ *     Clean up live restore metadata once background migration has completed. This will be called
+ *     by the last background migration thread. In most cases, we'll enter this function on
+ *     completion of background migration, but we also need to handle the case where we restart part
+ *     way though the clean up.
+ */
+static int
+__live_restore_clean_up(WT_SESSION_IMPL *session, WT_THREAD *ctx)
+{
+    WTI_LIVE_RESTORE_FS *lr_fs = (WTI_LIVE_RESTORE_FS *)S2C(session)->file_system;
+    const char *force_ckpt_cfg[] = {
+      WT_CONFIG_BASE(session, WT_SESSION_checkpoint), "force=true", NULL};
+
+    if (__wti_live_restore_get_state(session, lr_fs) ==
+      WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION) {
+        /*
+         * We need our metadata updates to be written out. Force a checkpoint on all trees before
+         * marking background migration complete. All empty extent lists must be written to the
+         * metadata durably first with the checkpoint.
+         */
+        WT_RET(__wt_checkpoint_db(ctx->session, force_ckpt_cfg, true));
+
+        uint64_t time_diff_ms;
+        WTI_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
+        __wt_timer_evaluate_ms(session, &server->start_timer, &time_diff_ms);
+        __wt_verbose(session, WT_VERB_LIVE_RESTORE_PROGRESS,
+          "Completed restoring %" PRIu64 " files in %" PRIu64 " seconds",
+          S2C(session)->live_restore_server->work_count, time_diff_ms / WT_THOUSAND);
+
+        WT_RET(__wti_live_restore_set_state(session, lr_fs, WTI_LIVE_RESTORE_STATE_CLEAN_UP));
```

Comment:
Right now this is guaranteed to enter the second if block, I assume you've done it this way with an intended future purpose in mind. Can we drop a FIXME in here?

Reply:
Or, this could be rewritten as a `switch` statement on the state, and have a `/* FALLTHROUGH */` comment to make sure it is clear. There are a bunch of uses/precedent for that in other places you can look at/copy.

Reply:
Thanks I went with the `switch` approach in https://github.com/wiredtiger/wiredtiger/pull/11590/commits/83b5714c773166ecf7698d9d65a82e6f0a41c6d7. 
We can enter this function in any of the `BACKGROUND_MIGRATION`/`CLEAN_UP`/`COMPLETE` stages so we sometimes need to skip the early, already run code (i.e. not running another checkpoint if we're already in the `CLEAN_UP` stage).